externs/browser/html5.js

```diff
@@ -2164,12 +2164,45 @@ MessagePort.prototype.onmessage;
 
 // HTML5 MessageEvent class
 /**
- * @see http://dev.w3.org/html5/spec/comms.html#messageevent
+ * @typedef {Window|MessagePort|ServiceWorker}
+ * @see https://html.spec.whatwg.org/multipage/comms.html#messageeventsource
+ */
+var MessageEventSource;
+
+
+/**
+ * @record
+ * @extends {EventInit}
+ * @template T
+ * @see https://html.spec.whatwg.org/multipage/comms.html#messageeventinit
+ */
+function MessageEventInit() {}
+
+/** @type {T|undefined} */
+MessageEventInit.prototype.data;
+
+/** @type {(string|undefined)} */
+MessageEventInit.prototype.origin;
+
+/** @type {(string|undefined)} */
+MessageEventInit.prototype.lastEventId;
+
+/** @type {(?MessageEventSource|undefined)} */
+MessageEventInit.prototype.source;
+
+/** @type {(!Array<MessagePort>|undefined)} */
+MessageEventInit.prototype.ports;
+
+
+/**
+ * @see https://html.spec.whatwg.org/multipage/comms.html#messageevent
  * @constructor
  * @extends {Event}
+ * @param {string} type
+ * @param {MessageEventInit<T>=} opt_eventInitDict
  * @template T
  */
-function MessageEvent() {}
+function MessageEvent(type, opt_eventInitDict) {}
```

Comment:
I would prefer if we could try to land this because calling `MessageEvent` without any argument throws `Uncaught TypeError: Failed to construct 'MessageEvent': 1 argument required, but only 0 present.` in Chrome, Firefox and Safari.